mocha/reporter.js

```diff
@@ -0,0 +1,104 @@
+const { v4: uuidv4 } = require('uuid')
+const CI = require('../util/ci')
+const Paths = require('../util/paths')
+const Mocha = require('mocha')
+const Runnable = require('mocha/lib/runnable')
+const uploadTestResults = require('../util/uploadTestResults')
+const failureExpanded = require('../util/failureExpanded')
+
+const {
+  EVENT_RUN_END,
+  EVENT_TEST_BEGIN,
+  EVENT_TEST_END,
+  EVENT_TEST_PENDING,
+} = Mocha.Runner.constants
+
+const {
+  STATE_PASSED,
+  STATE_PENDING,
+  STATE_FAILED,
+} = Runnable.constants
+
+class MochaBuildkiteAnalyticsReporter {
+  constructor(runner) {
+    this._testResults = []
+    this._testEnv = (new CI()).env();
+    this._paths = new Paths({ cwd: process.cwd() }, this._testEnv.location_prefix)
+
+    runner
+      .on(EVENT_TEST_BEGIN, (test) => {
+        this.testStarted(test)
+      })
+      .on(EVENT_TEST_END, (test) => {
+        this.testFinished(test)
+      })
+      .on(EVENT_RUN_END, () => {
+        this.testRunFinished()
+      })
+  }
+
+  testStarted(test) {
+    test.testAnalyticsId = uuidv4()
+    test.startAt = performance.now() / 1000
+  }
+
+  testFinished(test) {
+    const failureReason = test.err !== undefined ? test.err.toString() : undefined
+    const prefixedTestPath = this._paths.prefixTestPath(test.file)
```

Comment:
It could be a Cypress specific issue, closes I could find is [this](https://github.com/cypress-io/cypress/issues/1495), but that's severely outdated and I am using cypress 8.7.0. Pretty much the exact same structure you just sent. Tomorrow I'll do some further testing and research on it